src/components/Article/ArticlesHomepage.tsx

```diff
@@ -36,23 +37,42 @@ interface Article {
 
 const ArticlesHomepage: React.FC = () => {
   const [articles, setArticles] = useState<Article[]>([]);
+  const [isUserSignedIn, MakeUserSignin] = useState<boolean>(false);
   const [isLoading, setIsLoading] = useState<boolean>(true);
   const [likedArticles, setLikedArticles] = useState<string[]>([]);
 
   useEffect(() => {
-    if (localStorage.getItem("userUid") == null) {
-      window.location.href = "#/";
-    }
     const fetchArticles = async () => {
       const articlesRef = ref(database, "articles");
       const userUid = localStorage.getItem("userUid");
 
       if (!userUid) {
+        try {
+          const snapshot = await get(articlesRef);
```

Comment:
Why have you added this? Please explain.

Reply:
I added it so we can fetch articles to show if user is not signed in